server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -674,7 +678,22 @@ public void setLastAcceptedState(ClusterState clusterState) {
             try {
                 final ClusterMetadataManifest manifest;
                 if (shouldWriteFullClusterState(clusterState)) {
-                    manifest = remoteClusterStateService.writeFullMetadata(clusterState);
+                    if (clusterState.metadata().clusterUUIDCommitted() == true) {
+                        final Optional<ClusterMetadataManifest> latestManifest = remoteClusterStateService.getLatestClusterMetadataManifest(
+                            clusterState.getClusterName().value(),
+                            clusterState.metadata().clusterUUID()
+                        );
+                        if (latestManifest.isPresent()) {
+                            previousClusterUUID = latestManifest.get().getPreviousClusterUUID();
+                        } else {
+                            logger.error(
+                                "Latest manifest is not present in remote store for cluster UUID: {}",
+                                clusterState.metadata().clusterUUID()
+                            );
+                            previousClusterUUID = ClusterState.UNKNOWN_UUID;
```

Comment:
when will this happen?

Reply:
When the user starts the cluster with remote state disabled. Then later enables the remote state and performs a full restart, there will not be any manifest for the current cluster UUID